pkg/metareconciler/sample/reconciler.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sample
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/logging"
+)
+
+type Reconciler struct {
+	lister cache.GenericLister
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
```

Comment:
Just curious why this is not using the ReconcileKind? Seems like that pattern has been quite useful and this seems like a step backwards if you have to add back all the boilerplating, handling deletions, etc.